frontend/src/lib/components/LemonSelectWithSearch/LemonSelectWithSearch.tsx

```diff
@@ -0,0 +1,97 @@
+import { Select } from 'antd'
+import React from 'react'
+import { UserBasicType, UserType } from '~/types'
+import { LemonSnack } from '../LemonSnack/LemonSnack'
+import { ProfilePicture } from '../ProfilePicture'
+import './LemonSelectWithSearch.scss'
+
+export interface LemonSelectWithSearchOption {
+    label: string | React.ReactNode
+    disabled?: boolean
+    'data-attr'?: string
+}
+
+export interface LemonSelectWithSearchOptionItem extends LemonSelectWithSearchOption {
+    key: string
+}
+
+export type LemonSelectWithSearchOptions = Record<string, LemonSelectWithSearchOption>
+
+export interface LemonSelectWithSearchProps {
+    options?: LemonSelectWithSearchOptions | LemonSelectWithSearchOptionItem[]
+    value?: string[] | null
+    disabled?: boolean
+    loading?: boolean
+    placeholder?: string
+    onChange?: (newValue: string[]) => void
+    onSearch?: (value: string) => void
+    filterOption?: boolean
+    mode?: 'single' | 'multiple' | 'multiple-custom'
+    'data-attr'?: string
+}
+
+export function LemonSelectWithSearch({
```

Comment:
Hmm, I'm not sure `LemonSelectWithSearch` is the right way to put this. If search is the issue, it should be supported as an extra feature of `LemonSelect`. However, it seems the primary motivation behind adding `LemonSelectWithSearch` is selection of multiple options. In that case, I advocate for naming this `LemonSelectMultiple`. The API should be practically the same as `LemonSelect`'s, only the selected values state should be an array.

Reply:
Funnily enough that was my original name for it and then I changed it to match the other `LemonButton` namings (gotta love me some `LemonButtonWithSideAction`... 😉

Seems like we're on the same page then